oga/backend/users/views.py

```diff
@@ -1,42 +1,68 @@
 from django.shortcuts import render, get_object_or_404
-from django.http import HttpResponse, HttpResponseBadRequest, JsonResponse
+from django.http import HttpResponse, HttpResponseBadRequest, JsonResponse, HttpResponseNotAllowed
 from django.views.decorators.csrf import csrf_exempt
 import json
 from .models import Profile, Question, Answer, Location
 from django.contrib.auth.models import User
 from django.contrib.auth import login, authenticate, get_user
 from django.views import generic
+from functools import wraps
 
+def login_required(views_func):
+    @wraps(views_func)
+    def wrapper(request, *args, **kwargs):
+        if request.user.is_authenticated:
+            return views_func(request, *args, **kwargs)
+        else:
+            return HttpResponse(status=400)
+    return wrapper
+
+def check_request(views_func):
+    @wraps(views_func)
+    def wrapper(request, *args, **kwargs):
+        try:
+            return views_func
+        except (KeyError, ValueError) as e:
+            return HttpResponse(status=400)
```

Comment:
I think using something like
```python
return HttpResponseBadRequest(str(e))
```
might be better for understanding code.

Reply:
thanks. I changed the code as you said